### ISSUE TITLE
perl calls [] "anonymous list" instead of array

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14231]
Migrated from [rt.perl.org#123182](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123182) (status was 'resolved')

Searchable as RT123182$

**Comment by p5pRT** [comment]
### From @mauke

#### Created by @mauke



$ perl \-we '\[\]'
Useless use of anonymous list \(\[\]\) in void context at \-e line 1\.

I don't like the "anonymous list" bit\. \[\] constructs a reference to an
anonymous array\. \[\] is a circumfix operator \*surrounding\* a list\, but the thing
in void context is not a list \(besides\, how can you have a list in a non\-list
context?\)\.

I think this op should be renamed to "anonymous array"\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.20.1:

Configured by mauke at Thu Nov  6 21:08:16 CET 2014.

Summary of my perl5 (revision 5 version 20 subversion 1) configuration:
   
  Platform:
    osname=linux, osvers=3.17.1-1-arch, archname=i686-linux
    uname='linux simplicio 3.17.1-1-arch #1 smp preempt wed oct 15 15:36:07 cest 2014 i686 gnulinux '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.9.1 20140903 (prerelease)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/gcc/i686-pc-linux-gnu/4.9.1/include-fixed /usr/lib /lib
    libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.20.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.20'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector'



@INC for perl 5.20.1:
    /home/mauke/usr/lib/perl5/site_perl/5.20.1/i686-linux
    /home/mauke/usr/lib/perl5/site_perl/5.20.1
    /home/mauke/usr/lib/perl5/5.20.1/i686-linux
    /home/mauke/usr/lib/perl5/5.20.1
    .


Environment for perl 5.20.1:
    HOME=/home/mauke
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LC_COLLATE=POSIX
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/mauke/perl5/perlbrew/bin:/home/mauke/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
    PERLBREW_BASHRC_VERSION=0.69
    PERLBREW_HOME=/home/mauke/.perlbrew
    PERLBREW_ROOT=/home/mauke/perl5/perlbrew
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @mauke

On Tue Nov 11 12&#8203;:19&#8203;:12 2014\, mauke\- wrote&#8203;:
> I think this op should be renamed to "anonymous array"\.

Patch attached\.

**Comment by p5pRT** [comment]
### From @mauke

<details><summary>0001-rename-from-anonymous-list-to-anonymous-array.patch</summary>

```diff
From b8299c879bfa967e2a0ac8a1a884babdc0b1b7ac Mon Sep 17 00:00:00 2001
From: Lukas Mai <l.mai@web.de>
Date: Tue, 11 Nov 2014 15:45:43 +0100
Subject: [PATCH] rename [] from "anonymous list" to "anonymous array"

---
 opcode.h            | 2 +-
 regen/opcodes       | 2 +-
 t/op/postfixderef.t | 2 +-
 t/op/ref.t          | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/opcode.h b/opcode.h
index 51fbf28..105dcbf 100644
--- a/opcode.h
+++ b/opcode.h
@@ -690,7 +690,7 @@ EXTCONST char* const PL_op_desc[] = {
 	"join or string",
 	"list",
 	"list slice",
-	"anonymous list ([])",
+	"anonymous array ([])",
 	"anonymous hash ({})",
 	"splice",
 	"push",
diff --git a/regen/opcodes b/regen/opcodes
index d6063a4..d3da201 100644
--- a/regen/opcodes
+++ b/regen/opcodes
@@ -246,7 +246,7 @@ join		join or string		ck_join		fmst@	S L
 
 list		list			ck_null		m@	L
 lslice		list slice		ck_null		2	H L L
-anonlist	anonymous list ([])	ck_fun		ms@	L
+anonlist	anonymous array ([])	ck_fun		ms@	L
 anonhash	anonymous hash ({})	ck_fun		ms@	L
 
 splice		splice			ck_fun		m@	A S? S? L
diff --git a/t/op/postfixderef.t b/t/op/postfixderef.t
index 3d60b9e..79b66de 100644
--- a/t/op/postfixderef.t
+++ b/t/op/postfixderef.t
@@ -96,7 +96,7 @@ $x = "Good";
 is ($refref->$*->$*, 'Good'); # is ($$$refref, 'Good');
 
 
-# Test nested anonymous lists.
+# Test nested anonymous arrays.
 
 $ref = [[],2,[3,4,5,]];
 is (scalar $ref->@*, 3); # is (scalar @$ref, 3);
diff --git a/t/op/ref.t b/t/op/ref.t
index 6ce0480..581aa71 100644
--- a/t/op/ref.t
+++ b/t/op/ref.t
@@ -80,7 +80,7 @@ $refref = \\$x;
 $x = "Good";
 is ($$$refref, 'Good');
 
-# Test nested anonymous lists.
+# Test nested anonymous arrays.
 
 $ref = [[],2,[3,4,5,]];
 is (scalar @$ref, 3);
-- 
2.1.3


```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Nov 11 12&#8203;:23&#8203;:58 2014\, mauke\- wrote&#8203;:
> On Tue Nov 11 12&#8203;:19&#8203;:12 2014\, mauke\- wrote&#8203;:
> > I think this op should be renamed to "anonymous array"\.
> 
> Patch attached\.

Thank you\.  Applied as 26321c2d1\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'pending release'

**Comment by p5pRT** [comment]
### From madcityzen@gmail.com


> On Nov 11\, 2014\, at 3&#8203;:02 PM\, Father Chrysostomos via RT \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> 
> On Tue Nov 11 12&#8203;:23&#8203;:58 2014\, mauke\- wrote&#8203;:
>> On Tue Nov 11 12&#8203;:19&#8203;:12 2014\, mauke\- wrote&#8203;:
>>> I think this op should be renamed to "anonymous array"\.
>> 
>> Patch attached\.
> 
> Thank you\.  Applied as 26321c2d1\.

Here are a couple more

\`ack '\(?\<=anon\)\\s\*list|\(?\<=anonymous\)\\s\*list'\` shows others\, but the op is pp\_anonlist\, so that shows up too\.



---
From 0c246d0bb1f71eec0c2adf48f1a5c774ccb923e1 Mon Sep 17 00&#8203;:00&#8203;:00 2001
From&#8203;: Doug Bell \<madcityzen@&#8203;gmail\.com>
Date&#8203;: Tue\, 11 Nov 2014 15&#8203;:19&#8203;:45 \-0600
Subject&#8203;: \[PATCH\] rename anonymous list \-> array in docs



---
 dist/Exporter/lib/Exporter\.pm | 2 \+\-
 lib/B/Deparse\.t               | 2 \+\-
 lib/Unicode/UCD\.pm            | 4 \+\+\-\-
 3 files changed\, 4 insertions\(\+\)\, 4 deletions\(\-\)

<details><summary>Inline Patch</summary>

```diff
diff --git a/dist/Exporter/lib/Exporter.pm b/dist/Exporter/lib/Exporter.pm
index bc72c51..e1e4e4a 100644
--- a/dist/Exporter/lib/Exporter.pm
+++ b/dist/Exporter/lib/Exporter.pm
@@ -223,7 +223,7 @@ right. Specifications are in the form:
 
     [!]name         This name only
     [!]:DEFAULT     All names in @EXPORT
-    [!]:tag         All names in $EXPORT_TAGS{tag} anonymous list
+    [!]:tag         All names in $EXPORT_TAGS{tag} anonymous array
     [!]/pattern/    All names in @EXPORT and @EXPORT_OK which match
 
 A leading ! indicates that matching names should be deleted from the
diff --git a/lib/B/Deparse.t b/lib/B/Deparse.t
index d05e3af..75255cc 100644
--- a/lib/B/Deparse.t
+++ b/lib/B/Deparse.t
@@ -1559,7 +1559,7 @@ state($s3, $s4);
 #@z = ($s7, undef, $s8);
 ($s7, undef, $s8) = (1, 2, 3);
 ####
-# anon lists with padrange
+# anon arrays with padrange
 my($a, $b);
 my $c = [$a, $b];
 my $d = {$a, $b};
diff --git a/lib/Unicode/UCD.pm b/lib/Unicode/UCD.pm
index 8632a3f..d7e1c86 100644
--- a/lib/Unicode/UCD.pm
+++ b/lib/Unicode/UCD.pm
@@ -612,7 +612,7 @@ See also L</Blocks versus Scripts>.
 If supplied with an argument that can't be a code point, C<charblock()> tries to
 do the opposite and interpret the argument as an old-style block name.  On an
 ASCII platform, the return value is a I<range set> with one range: an
-anonymous list with a single element that consists of another anonymous list
+anonymous array with a single element that consists of another anonymous array
 whose first element is the first code point in the block, and whose second
 element is the final code point in the block.  On an EBCDIC
 platform, the first two Unicode blocks are not contiguous.  Their range sets
@@ -720,7 +720,7 @@ that it doesn't have scripts, this function returns C<"Unknown">.
 
 If supplied with an argument that can't be a code point, charscript() tries
 to do the opposite and interpret the argument as a script name. The
-return value is a I<range set>: an anonymous list of lists that contain
+return value is a I<range set>: an anonymous array of arrays that contain
 I<start-of-range>, I<end-of-range> code point pairs. You can test whether a
 code point is in a range set using the L</charinrange()> function.
 (To be precise, each I<range set> contains a third array element,
-- 
2.0.0
```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Nov 11 13&#8203;:34&#8203;:40 2014\, madcityzen@&#8203;gmail\.com wrote&#8203;:
> 
> > On Nov 11\, 2014\, at 3&#8203;:02 PM\, Father Chrysostomos via RT \<perlbug\-
> > followup@&#8203;perl\.org> wrote&#8203;:
> >
> > On Tue Nov 11 12&#8203;:23&#8203;:58 2014\, mauke\- wrote&#8203;:
> >> On Tue Nov 11 12&#8203;:19&#8203;:12 2014\, mauke\- wrote&#8203;:
> >>> I think this op should be renamed to "anonymous array"\.
> >>
> >> Patch attached\.
> >
> > Thank you\.  Applied as 26321c2d1\.
> 
> Here are a couple more
> 
> \`ack '\(?\<=anon\)\\s\*list|\(?\<=anonymous\)\\s\*list'\` shows others\, but the
> op is pp\_anonlist\, so that shows up too\.
> 
> \-\-\-
> From 0c246d0bb1f71eec0c2adf48f1a5c774ccb923e1 Mon Sep 17 00&#8203;:00&#8203;:00 2001
> From&#8203;: Doug Bell \<madcityzen@&#8203;gmail\.com>
> Date&#8203;: Tue\, 11 Nov 2014 15&#8203;:19&#8203;:45 \-0600
> Subject&#8203;: \[PATCH\] rename anonymous list \-> array in docs

Thank you\.  Applied as df2b00e846\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From @ikegami

On Tue\, Nov 11\, 2014 at 3&#8203;:19 PM\, l\.mai@&#8203;web\.de \<perlbug\-followup@&#8203;perl\.org>
wrote&#8203;:

> \(besides\, how can you have a list in a non\-list context?\)\.
>

Easy\.

$ perl \-MO=Concise\,\-exec \-e'$x = \( f\(\)\, g\(\) \);'
1  \<0> enter
2  \<;> nextstate\(main 1 \-e&#8203;:1\) v&#8203;:\{
3  \<0> pushmark v
4  \<0> pushmark s
5  \<\#> gv\[\*f\] s/EARLYCV
6  \<1> entersub\[t4\] vKS/TARG
7  \<0> pushmark s
8  \<\#> gv\[\*g\] s/EARLYCV
9  \<1> entersub\[t7\] sKS/TARG
a  \<@&#8203;> list sKP                           \<\-\-\- List in \(s\)calar context
b  \<\#> gvsv\[\*x\] s
c  \<2> sassign vKS/2
d  \<@&#8203;> leave\[1 ref\] vKP/REFC
\-e syntax OK

$ perl \-MO=Concise\,\-exec \-e'f\(\)\, g\(\)'
1  \<0> enter
2  \<;> nextstate\(main 1 \-e&#8203;:1\) v&#8203;:\{
3  \<0> pushmark v
4  \<0> pushmark s
5  \<\#> gv\[\*f\] s/EARLYCV
6  \<1> entersub\[t3\] vKS/TARG
7  \<0> pushmark s
8  \<\#> gv\[\*g\] s/EARLYCV
9  \<1> entersub\[t6\] vKS/TARG
a  \<@&#8203;> list vK                            \<\-\-\- List in \(v\)oid context
b  \<@&#8203;> leave\[1 ref\] vKP/REFC
\-e syntax OK

A sub can't return a list in non\-list context\, and an operator can't
evaluate to a list in non\-list context\, but you can have a list operator in
scalar context\.


> I think this op should be renamed to "anonymous array"\.
>

Definitely\. It's clearly referring to an array\, not any accepted definition
of list\.

**Comment by p5pRT** [comment]
### From @khwilliamson

Thanks for submitting this ticket

The issue should be resolved with the release today of Perl v5\.22\, available at http&#8203;://www\.perl\.org/get\.html
If you find that the problem persists\, feel free to reopen this ticket

\-\- 
Karl Williamson for the Perl 5 porters team

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'pending release' to 'resolved'
